dateparser/freshness_date_parser.py

```diff
@@ -111,21 +113,34 @@ def apply_time(dateobj, timeobj):
         return date, period
 
     def _parse_date(self, date_string, prefer_dates_from):
-        if not self._are_all_words_units(date_string):
+
+        _weekday = self.get_weekday_data(date_string)
+
+        if not self._are_all_words_units(date_string) and not _weekday:
             return None, None
 
         kwargs = self.get_kwargs(date_string)
-        if not kwargs:
+
+        if not kwargs and not _weekday:
             return None, None
 
         period = 'day'
-        if 'days' not in kwargs:
-            for k in ['weeks', 'months', 'years']:
-                if k in kwargs:
-                    period = k[:-1]
-                    break
+        if _weekday:
+            day = "cal." + _weekday.upper()
+            day_ahead = eval(day) - datetime.now().weekday()
```

Comment:
Won’t there be issues if timezones are involved?

Reply:
All tests are passing still if we try something like  
```
>>> dateparser.parse('next monday at 10 pm UTC-2')
datetime.datetime(2020, 3, 30, 14, 22, 45, 602138, tzinfo=<StaticTzInfo 'UTC\-02:00'>)
```
it fails to include the correct time but timezone is correct.

Any suggestions on this?

Reply:
I already assumed we cannot scrape timezones in weekday name strings at the moment, and I don’t have a problem with that, it can be improved later if we find real life usecases worth supporting.

The scenario I’m concerned about is: user supplies a timezone through settings (there is a `TIMEZONE` setting), user parses “Next Monday”, and the current day is determined based on UTC, not the user-supplied timezone, which may cause today to be a different day than expected, and hence affect the parsing of “Next Monday” by 1 week.

Reply:
yeah, I understand your point now.